src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -705,33 +713,76 @@ const WorkflowView = _.flow(
               div([
                 div({ style: { height: '2rem', fontWeight: 'bold' } }, ['Step 1']),
                 label(['Select root entity type:']),
-                h(Select, {
+                h(GroupedSelect, {
                   'aria-label': 'Entity type selector',
                   isClearable: false,
                   isDisabled: currentSnapRedacted || this.isSingle() || !!Utils.editWorkspaceError(ws),
                   isSearchable: true,
                   placeholder: 'Select data type...',
                   styles: { container: old => ({ ...old, display: 'inline-block', width: 200, marginLeft: '0.5rem' }) },
                   value: selectedEntityType,
-                  onChange: selection => {
-                    const value = this.updateEntityType(selection)
-                    this.setState({ entitySelectionModel: this.resetSelectionModel(value) })
+                  onChange: async ({ value, source }) => {
+                    if (source === 'snapshot') {
+                      const selectedSnapshotCachedMetadata = _.find({ snapshotName: value }, cachedSnapshotEntityMetadata)
```

Comment:
TOL: If we're just doing plain caching here, it might be simpler to use something like `memoizeAsync` instead of managing the cache by hand.